### PR TITLE
Add missing SET_OF_encode_uper

### PR DESCRIPTION
**Pull request by craig65535**
This is based on the patch referenced in #42

### CODE REVIEW

**Comment by vlm** [comment]
This is not a correct patch for CANONICAL Unaligned PER.

X.691 (08/2015) #22.1 encoding of the component values of the set-of type shall appear in ascending
order”.

**Comment by craig65535** [comment]
@vlm Updated. Please let me know if I'm on the right track

File: skeletons/constr_SET_OF.c [hunk 1]
```diff
@@ -984,26 +1049,59 @@ SET_OF_encode_uper(const asn_TYPE_descriptor_t *td,
...
+
+	/* build an encoded list */
+	for(seq = 0; seq < list->count; seq++) {
+		void *memb_ptr = list->array[seq];
+		if(!memb_ptr) {
+			_uper_encoded_list_free(encoded_list, num_encoded);
+			ASN__ENCODE_FAILED;
+		}
+		encoded_list[seq].seq = seq;
+		er = uper_encode(elm->type, memb_ptr, _uper_out, &encoded_list[seq]);
```
Comment (vlm) [inline skeletons/constr_SET_OF.c]: This ignores possible constraint on elm->type.

File: skeletons/constr_SET_OF.c [hunk 2]
```diff
@@ -984,26 +1049,59 @@ SET_OF_encode_uper(const asn_TYPE_descriptor_t *td,
...
 
 		while(mayEncode--) {
-			void *memb_ptr = list->array[seq++];
-			if(!memb_ptr) ASN__ENCODE_FAILED;
+			void *memb_ptr = list->array[encoded_list[seq++].seq];
+			if(!memb_ptr) {
+				_uper_encoded_list_free(encoded_list, num_encoded);
+				ASN__ENCODE_FAILED;
+			}
 			er = elm->type->op->uper_encoder(elm->type,
```
Comment (vlm) [inline skeletons/constr_SET_OF.c]: The data is already there, in the encoded list. Why not use it instead.

**Comment by vlm** [comment]
Thank you for the prompt, I gave it a whirl myself. Checked into the master, together with randomized tests to verify correctness.

**Comment by vlm** [comment]
Implemented as a separate effort.

**Pull request closed without merging**